data_labeling_examples/src/main/java/com/oracle/datalabelingservicesamples/requests/Config.java

```diff
@@ -0,0 +1,135 @@
+package com.oracle.datalabelingservicesamples.requests;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.oracle.bmc.ConfigFileReader;
+import com.oracle.bmc.auth.AuthenticationDetailsProvider;
+import com.oracle.bmc.auth.ConfigFileAuthenticationDetailsProvider;
+import com.oracle.bmc.datalabelingservicedataplane.DataLabelingClient;
+import com.oracle.datalabelingservicesamples.labelingstrategies.CustomLabelMatch;
+import com.oracle.datalabelingservicesamples.labelingstrategies.FirstLetterMatch;
+import com.oracle.datalabelingservicesamples.labelingstrategies.FirstRegexMatch;
+import com.oracle.datalabelingservicesamples.labelingstrategies.LabelingStrategy;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Getter
+@Slf4j
+public enum Config {
+
+	INSTANCE;
+
+	private DataLabelingClient dlsDpClient;
+	private String configFilePath;
+	private String configProfile;
+	private String dpEndpoint;
+	private String datasetId;
+	private String region;
+
+	private List<String> labels;
+	private Map<String, List<String>> customLabels;
+	private String labelingAlgorithm;
+	private LabelingStrategy labelingStrategy;
+	private String regexPattern;
+	private int threadCount;
+
+	private Config() {
+		try {
+			Properties config = new Properties();
+			config.load(getClass().getClassLoader().getResourceAsStream("config.properties"));
+			configFilePath = config.getProperty("CONFIG_FILE_PATH");
+			configProfile = config.getProperty("CONFIG_PROFILE");
+			dpEndpoint = config.getProperty("DLS_DP_URL");
+			datasetId = config.getProperty("DATASET_ID");
+			region = config.getProperty("REGION");
+			labelingAlgorithm = config.getProperty("LABELING_ALGORITHM");
+			String threadConfig = config.getProperty("THREAD_COUNT");
```

Comment:
Probably a rare edge case scenario, but does this need any validation? For example, what if a very high number or a negative no is configured for threadConfig?

Reply:
Added.